### PR TITLE
fix(form): asterisks ::before pseudoclass

### LINKED ISSUE #104: fix(form): asterisks before labels with errors
the asterisks should be handled by css using the :before pseudoclass and not conditionally added via the template; this makes it easy to "theme" later (ie, we can have icons instead of asterisks later)

### PR DESCRIPTION
**Pull request by Icyscools**
## fix(form): asterisks `::before` pseudoclass
Fixes #104 

Description of changes: 
- Remove a display asterisk condition in label, replacing with a `::before` pseudoclass.

Tag a reviewer: @ayoayco 

Tasks:
- [ ] I have ran the build command to make sure apps are working: `npm run build`
- [x] I have ran the tests to make sure nothing is broken: `npm run test`
- [x] I have ran the linter to make sure code is clean: `npm run lint`
- [x] I have reviewed my own code to remove changes that are not needed

### CODE REVIEW

File: packages/form/components/Field.astro [hunk 1]
```diff
@@ -95,4 +95,8 @@ const validatorAttributes: Record<string, string> = validators?.reduce(
 		color: red;
 		border-color: red;
 	}
+
+	div[data-required='true'] + label::before {
```
Comment (ayoayco) [inline packages/form/components/Field.astro:99]: I think it should be `data-validator-required` can you check?

**Review by ayoayco — CHANGES_REQUESTED** [review]

File: packages/form/components/Field.astro (hunk 1 above)
Comment (Icyscools) [reply packages/form/components/Field.astro]: Sure, fixed!

**Comment by Icyscools** [comment]
I notice that I didn't pass the test in `Field.astro` since I change the method to render the asterisk, so I think we have to change the test later.

**Comment by ayoayco** [comment]
We do not want the main branch to fail the build and test pipeline. So maybe you have to update the test, as long as it is an accurate assertion. Do you see asterisks on the demo app but the tests are still failing?

**Comment by Icyscools** [comment]
Yes, I see the asterisk on the demo page but when I ran tests it failed.

I agreed that we have to keep the main branch should passed all the tests, so I'll check on it.

**Comment by ayoayco** [comment]
Let me know what you find. I don't mind skipping the test for now, maybe the testing library we are using for astro is still not able to render css changes.

**Comment by Icyscools** [comment]
I investigate the issue on tests and discover that the testing library we're currently using cannot parse displaying from CSS. 

As for now, I've changed the test to check on data-attribute rather than an asterisk when the input field is required.

**Review by ayoayco — APPROVED** [review]
Test change is indeed better than checking for asterisks. Good job 🚀

**Pull request merged**